### PR TITLE
CDAP-4975 Fix HA RM delegation token format

### PR DESCRIPTION
**Pull request by poornachandra**
CDAP-4767 Give every application its own delegation tokens so that cancellation of delegation tokens of one application does not affect another application

Build - http://builds.cask.co/browse/CDAP-RBT644-3

### CODE REVIEW

File: cdap-common/src/main/java/org/apache/twill/internal/yarn/Hadoop21YarnAppClient.java [hunk 1]
```diff
@@ -0,0 +1,167 @@
...
+
+import java.util.List;
+import javax.annotation.Nullable;
+
+/**
+ * <p>
+ * The service implementation of {@link YarnAppClient} for Apache Hadoop 2.1 and beyond.
+ *
+ * The {@link VersionDetectYarnAppClientFactory} class will decide to return instance of this class for
+ * Apache Hadoop 2.1 and beyond.
```
Comment (chtyim) [inline cdap-common/src/main/java/org/apache/twill/internal/yarn/Hadoop21YarnAppClient.java]: Add a TODO to get this class removed when the underlying problem is fixed in YARN (TWILL-119). The TODO can refer to CDAP-4923 for the upgrade to twill 0.8.0

**Comment by chtyim** [comment]
Just one comment about adding a TODO. Otherwise LGTM.

**Comment by sreevatsanraman** [comment]
lgtm

**Comment by poornachandra** [comment]
Squashing commits.

**Comment by poornachandra** [comment]
Merging since last test failure was an unrelated one.

**Pull request merged**